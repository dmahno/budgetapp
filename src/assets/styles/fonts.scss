@font-face {
  font-family: 'Golos';
  src:
    url('../fonts/Golos-Regular.woff2') format('woff2'),
    url('../fonts/fonts/Golos-Regular.ttf') format('truetype');
  font-weight: 400;
  font-style: normal;
}

@font-face {
  font-family: 'Golos';
  src:
    url('../fonts/fonts/Golos-Medium.woff2') format('woff2'),
    url('../fonts/fonts/Golos-Medium.ttf') format('truetype');
  font-weight: 500;
  font-style: normal;
}

@font-face {
  font-family: 'Golos';
  src:
    url('../fonts/fonts/Golos-SemiBold.woff2') format('woff2'),
    url('../fonts/fonts/Golos-SemiBold.ttf') format('truetype');
  font-weight: 600;
  font-style: normal;
}

@font-face {
  font-family: 'Golos';
  src:
    url('../fonts/fonts/Golos-Bold.woff2') format('woff2'),
    url('../fonts/fonts/Golos-Bold.ttf') format('truetype');
  font-weight: 700;
  font-style: normal;
}

@font-face {
  font-family: 'Golos';
  src:
    url('../fonts/fonts/GolosText-Black.woff2') format('woff2'),
    url('../fonts/fonts/GolosText-Black.ttf') format('truetype');
  font-weight: 800;
  font-style: normal;
}
