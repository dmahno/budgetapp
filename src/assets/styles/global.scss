@use 'assets/styles/fonts';
@use 'assets/styles/mobile';
@use 'assets/variables/colors';

* {
  box-sizing: border-box;
}

html,
body {
  margin: 0;
  padding: 0;
  size: 16px;
  font-family: 'Golos Text', serif;
  background: var(--backgroundColor);
  color: var(--textColor);
}

body {
  min-width: 320px;
  padding: 0 $indent-s;
}

@include mobile {
  body {
    font-size: 14px;
  }
}

h1,
h2,
h3,
h4,
h5,
h6,
p,
input,
button,
textarea,
ul,
li {
  margin: 0;
  padding: 0;
}

li {
  list-style-type: none;
}

.link,
.active {
  transition:
    background-color 0.2s ease,
    color 0.2s ease;
}

svg {
  fill: var(--iconColor);
}

a {
  color: var(--linkColor);
}

input:-webkit-autofill,
input:-webkit-autofill:focus {
  transition:
    background-color 0s 600000s,
    color 0s 600000s !important;
}

button {
  background-color: var(--buttonBackgroundColor);
  color: var(--buttonTextColor);
  border: none;
  padding: 10px 20px;
  cursor: pointer;
}

html {
  scroll-behavior: smooth;
}
