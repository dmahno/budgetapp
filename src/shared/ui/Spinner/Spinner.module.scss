$spinner-sizes: (
  s: 24px,
  m: 56px,
  l: 80px,
);

$light-color: #474bff;
$dark-color: #fff;

.spinner {
  border-radius: 50%;
  animation: spinnerAnimation 1s infinite linear;

  &.light {
    background:
      radial-gradient(farthest-side, $light-color 94%, transparent) top/9px 9px
        no-repeat,
      conic-gradient(transparent 30%, $light-color);
    -webkit-mask: radial-gradient(
      farthest-side,
      transparent calc(100% - 9px),
      #000 0
    );
  }

  &.dark {
    background:
      radial-gradient(farthest-side, $dark-color 94%, transparent) top/9px 9px
        no-repeat,
      conic-gradient(transparent 30%, $dark-color);
    -webkit-mask: radial-gradient(
      farthest-side,
      transparent calc(100% - 9px),
      #000 0
    );
  }

  &.s {
    width: map-get($spinner-sizes, s);
    height: map-get($spinner-sizes, s);

    &.light {
      background:
        radial-gradient(farthest-side, $light-color 94%, transparent) top/5px
          5px no-repeat,
        conic-gradient(transparent 20%, $light-color);
      -webkit-mask: radial-gradient(
        farthest-side,
        transparent calc(100% - 5px),
        #000 0
      );
    }

    &.dark {
      background:
        radial-gradient(farthest-side, $dark-color 94%, transparent) top/5px 5px
          no-repeat,
        conic-gradient(transparent 20%, $dark-color);
      -webkit-mask: radial-gradient(
        farthest-side,
        transparent calc(100% - 5px),
        #000 0
      );
    }
  }

  &.m {
    width: map-get($spinner-sizes, m);
    height: map-get($spinner-sizes, m);
  }

  &.l {
    width: map-get($spinner-sizes, l);
    height: map-get($spinner-sizes, l);
  }
}

@keyframes spinnerAnimation {
  100% {
    transform: rotate(1turn);
  }
}
