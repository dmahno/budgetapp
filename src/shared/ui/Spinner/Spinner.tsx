import React from 'react';
import classNames from 'classnames';

import styles from './Spinner.module.scss';

type TSpinnerSize = 's' | 'm' | 'l';
type TSpinnerMode = 'light' | 'dark';

interface ISpinnerProps {
  size: TSpinnerSize;
  mode: TSpinnerMode;
  className?: string;
}

export const Spinner: React.FC<Partial<ISpinnerProps>> = ({
  size = 'm',
  mode = 'light',
  className,
}) => {
  return (
    <div
      data-testid="spinner"
      className={classNames(
        styles.spinner,
        styles[size],
        styles[mode],
        className,
      )}
    />
  );
};
