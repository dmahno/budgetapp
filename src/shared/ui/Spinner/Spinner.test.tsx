import {render} from '@testing-library/react';
import {describe, it, expect} from 'vitest';

import {Spinner} from './Spinner';
import styles from './Spinner.module.scss';

describe('Spinner', () => {
  it('renders with default props', () => {
    const {container} = render(<Spinner />);
    const spinner = container.firstChild;

    expect(spinner).toHaveClass(styles.spinner, styles['m'], styles['light']);
    expect(container).toMatchSnapshot();
  });

  it('renders with custom size "s" and mode "dark"', () => {
    const {container} = render(<Spinner size="s" mode="dark" />);
    const spinner = container.firstChild;

    expect(spinner).toHaveClass(styles.spinner, styles['s'], styles['dark']);
    expect(container).toMatchSnapshot();
  });

  it('renders with custom size "l" and mode "light"', () => {
    const {container} = render(<Spinner size="l" mode="light" />);
    const spinner = container.firstChild;

    expect(spinner).toHaveClass(styles.spinner, styles['l'], styles['light']);
    expect(container).toMatchSnapshot();
  });

  it('applies additional className if provided', () => {
    const {container} = render(<Spinner className="custom-class" />);
    const spinner = container.firstChild;

    expect(spinner).toHaveClass(
      styles.spinner,
      styles['m'],
      styles['light'],
      'custom-class',
    );
    expect(container).toMatchSnapshot();
  });
});
