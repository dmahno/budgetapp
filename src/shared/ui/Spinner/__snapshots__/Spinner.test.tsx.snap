// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Spinner > applies additional className if provided 1`] = `
<div>
  <div
    class="_spinner_b6745b _m_b6745b _light_b6745b custom-class"
    data-testid="spinner"
  />
</div>
`;

exports[`Spinner > renders with custom size "l" and mode "light" 1`] = `
<div>
  <div
    class="_spinner_b6745b _l_b6745b _light_b6745b"
    data-testid="spinner"
  />
</div>
`;

exports[`Spinner > renders with custom size "s" and mode "dark" 1`] = `
<div>
  <div
    class="_spinner_b6745b _s_b6745b _dark_b6745b"
    data-testid="spinner"
  />
</div>
`;

exports[`Spinner > renders with default props 1`] = `
<div>
  <div
    class="_spinner_b6745b _m_b6745b _light_b6745b"
    data-testid="spinner"
  />
</div>
`;
